GRDBSnapshotTestingTests/Migrations.swift

```diff
@@ -123,4 +123,37 @@ enum Migrations {
             """)
         }
     }
+    
+    enum SQLiteKeywordTableNames: DBMigration {
+        static func migrate(db: Database) throws {
+            try db.create(table: "database") { t in
```

Comment:
The goal is to test that we can properly dump the content of tables that use fancy identifiers, which implies quoting them when we run SQL requests.

`database` and `filter` do not need any escaping, so they are not very interesting:

```
$ sqlite3 
sqlite> CREATE TABLE database(a);
sqlite> CREATE TABLE filter(a);
```

On the other hand, `table` is:

```
sqlite> CREATE TABLE table(a);
Error: near "table": syntax error
```

As for primary key columns that need escaping, you can try:

```
sqlite> CREATE TABLE "table"(primary TEXT PRIMARY KEY);
Error: near "primary": syntax error
```

Reply:
(Yeah, the `sqlite3` command-line tool is a fine REPL)

Reply:
Sanity check:

```
sqlite> CREATE TABLE "table"("primary" TEXT PRIMARY KEY);
sqlite> SELECT * FROM table ORDER BY primary;
Error: near "table": syntax error
sqlite> SELECT * FROM "table" ORDER BY primary;
Error: near "primary": syntax error
sqlite> SELECT * FROM "table" ORDER BY "primary";
sqlite>
```

Yeah, both `table` and `primary` need escaping when we run the dump request.

Reply:
You're right. I only checked that `SELECT * FROM transaction` gives error and assumed that all tables named after keywords (e.g. database and filter) will also throw error.

Reply:
OK. `transaction` is fine :-) We may just need to change the test so that at least one primary key contains a column that needs escaping. I apologize if this test already exists and I missed it. In this case we're good to go :-)

Reply:
I removed `database` and `filter` tables from tests and replaced `transaction` table with `table` table with `primary` primary key. I also added additional test for `table` table with composite primary key  `(primary, join)`.